lib/resolve.ts

```diff
@@ -108,6 +108,16 @@ export function resolveTask(
 		templateVars,
 	);
 
+	// inject arguments
+	task.args = {
+		...templateVars,
+		...{
+			BALENA_ARCH: architecture,
+			BALENA_MACHINE_NAME: deviceType,
+			TARGETPLATFORM: platform,
```

Comment:
What is `TARGETPLATFORM` and why is it not prefixed with  `BALENA_`?

That is, is it some docker-wide thing or specific to balena?

Reply:
Never mind, I just noticed this in your description for the PR:

> This also includes TARGETPLATFORM, which buildx injects:

Reply:
I actually just thought of `BUILDPLATFORM` which we should probably inject as well, or just drop both...

In any case this is just a nice-to-have :)